src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -102,6 +105,21 @@ public void addMessage(Message message) {
     return messagesInConversation;
   }
 
+  /** Access the current set of Messages sent by a specific user. */
+  public Queue<Message> getMessagesByOwner(UUID authorid) {
+
+    Queue<Message> messagesByOwner = new PriorityQueue<>(messages.size(), 
+                                                         new MessageTimeComparator());
+
+    for (Message message : messages) {
```

Comment:
I'd refactor this like so:

```
messages.removeIf(!message.getAuthorId().equals(authorId));
return new PriorityQueue<>(messages);
```

If you override the compareTo method as I suggested above then the ordering will be done during construction of the PQ.

Reply:
Just implemented all of your suggestions locally, but I think maybe I don't understand this one too well. After I implemented removeIf, it effectively removes all of the other messages that weren't sent by the user when I visit their profile, but this means all other messages get deleted from all the conversations. This shouldn't be like this right?

Reply:
Ah I read this too quickly. Create a new list and then do removeIf (removeIf works in place). So something like:

List<Message> authorMessages = new ArrayList<>(messages);
authorMessages.removeIf ...